### PR TITLE
v4.1.x: VERSION: make safe for C++ compilers

### PR DESCRIPTION
**Pull request by jsquyres**
When compiling the MPI C++ bindings on a case-insensitive filesystem,
we #include various STL header files.  At least some versions of Clang
on MacOS have "#include <version>" in their STL header files, which
ends up including the Open MPI VERSION file.

The Open MPI VERSION file is executed as a Bourne shell script during
configure.  We could rename VERSION --> VERSION.sh, but this would
touch a lot of places, and doesn't seem worth it (remember: the MPI
C++ bindings have been removed from the Open MPI master and upcoming
v5.0.0 release).

This commit therefore does something small/simple to make the existing
VERSION file also safe to #include as a C/C++ header file.  It
basically uses "#if 0" to comment out the main body of the file, and
then "#include_next <version>" to get the "real" <version> file.

As mentioned above, this issue does not exist on the master, so this
is a one-off commit on the v4.1.x branch (not a cherry pick from
master).

Thanks to @srpgilles for raising the issue.

Signed-off-by: Jeff Squyres <jsquyres@cisco.com>

Refs #9122

bot:notacherrypick

### CODE REVIEW

**Comment by jsquyres** [comment]
After this PR is merged, we should cherry pick it to v4.0.x.

**Review by bwbarrett — APPROVED** [review]

**Comment by rhc54** [comment]
You are right - that looks totally _weird_....
